dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/integration/RegistryDirectory.java

```diff
@@ -222,7 +223,14 @@ private void refreshInvoker(List<URL> invokerUrls) {
             if (invokerUrls.isEmpty()) {
                 return;
             }
-            Map<URL, Invoker<T>> newUrlInvokerMap = toInvokers(invokerUrls);// Translate url list to Invoker map
+            
+            // can't use local reference because this.urlInvokerMap might be accessed at isAvailable() by main thread concurrently.
+            Map<URL, Invoker<T>> oldUrlInvokerMap = null;
+            if (this.urlInvokerMap != null) {
+                oldUrlInvokerMap = new LinkedHashMap<>(this.urlInvokerMap.size());
```

Comment:
Please consider load factor. `this.urlInvokerMap.size() / 0.75f + 1` would reduce one time resize.

Reply:
do you mean setting the initial capacity greater than the maximum number of entries divided by the load factor (default: 0.75) to avoid rehash operations?

Reply:
```
            if (this.urlInvokerMap != null) {
                // the initial capacity should be set greater than the maximum number of entries divided by the load factor to avoid resizing.
                oldUrlInvokerMap = new LinkedHashMap<>(Math.round(1 + this.urlInvokerMap.size() / DEFAULT_LOAD_FACTOR));
                this.urlInvokerMap.forEach(oldUrlInvokerMap::put);
            }
```
pls review again